avocado/utils/network/interfaces.py

```diff
@@ -0,0 +1,323 @@
+
+        :param version: Address Family Version (4 or 6). This must be a integer
+                        and default is 4.
+        :return: IP address as string.
+        """
+        if version not in [4, 6]:
+            raise NWException("Version {} not supported".format(version))
+
+        try:
+            details = self._get_interface_details(version)
+            addr_info = details[0].get('addr_info')
+            if addr_info:
+                return [x.get('local') for x in addr_info]
+        except (NWException, IndexError):
+            msg = "Could not get ip addresses for {}".format(self.name)
+            log.debug(msg)
+            return []
+
+    def get_hwaddr(self):
+        """Get the Hardware Address (MAC) of this interface.
+
+        This method will try to get the address and if fails it will raise a
+        NWException.
+        """
+        cmd = "cat /sys/class/net/{}/address".format(self.name)
+        try:
+            return _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to get hw address: {}".format(ex))
+
+    def get_link_state(self):
+        """Method used to get the current link state of this interface.
+
+        This method will return 'up', 'down' or 'unknown', based on the
+        network interface state. Or it will raise a NWException if is
+        unable to get the interface state.
+        """
+        cmd = "cat /sys/class/net/{}/operstate".format(self.name)
+        try:
+            return _run_command(cmd, self.remote_session)
+        except CmdError as e:
+            msg = ('Failed to get link state. Maybe the interface is '
+                   'missing. {}'.format(e))
+            raise NWException(msg)
+
+    def get_mtu(self):
+        """Return the current MTU value of this interface.
+        This method will try to get the current MTU value, if fails will
+        raise a NWException.
+        """
+        try:
+            return self._get_interface_details()[0].get('mtu')
+        except (NWException, IndexError):
+            raise NWException("Could not get MUT value.")
+
+    def ping_check(self, peer_ip, count=2, options=None):
+        """This method will try to ping a peer address (IPv4 or IPv6).
+
+        You should provide a IPv4 or IPV6 that would like to ping. This
+        method will try to ping the peer and if fails it will raise a
+        NWException.
+
+        :param peer_ip: Peer IP address (IPv4 or IPv6)
+        :param count: How many packets to send. Default is 2
+        :param options: ping command options. Default is None
+        """
+        cmd = "ping -I {} {} -c {}".format(self.name, peer_ip, count)
+        if options is not None:
+            cmd = "{} {}".format(cmd, options)
+        try:
+            _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to ping: {}".format(ex))
+
+    def save(self, ipaddr, netmask):
```

Comment:
@beraldoleal 
Hi Beraldo,
I just have a small question.
In case a system uses NetworkManager maybe we should check it. And after changing the ifcfg-files to run the "nmcli" reload for it?

Reply:
@anascko, yes. We can do improvments here. But since that this was already part of the old code. I would suggest to open an issue and try to solve this on a different PR.